### ISSUE TITLE
How to use?

### ISSUE THREAD

**Issue by janusch** [monzo/envoy-preflight#1]
Hi,
thank you for open sourcing this script.

I am pretty new to the container world. Could you eli5 the installation process of this?

Do I need to compile this before i can add it to my Dockerfile? 

I am interested in envoy-preflight because of the problem that the sidecar does not shut down and a kubernetes job never completes due to that.

Thank you again!

**Comment by jackkleeman** [comment]
Hi @janusch, sorry for my slow response

Here's an example dockerfile:
```Dockerfile
FROM scratch
ADD your-binary /
ADD envoy-preflight /
ENTRYPOINT ["/envoy-preflight", "/your-binary"]
```

And then in your makefile, something like:
```Makefile
envoy-preflight:
	go build -o envoy-preflight github.com/monzo/envoy-preflight

docker: your-binary envoy-preflight
	docker build  .

your-binary:
  # your build step goes here
```

You'll end up with a docker container where the entrypoint is envoy preflight, which will then call your binary. You can do something like:
```shell
docker run -it yourbuiltimage yourargs
```
and it should run your binary, passing the args (if you need any args).

Generally a docker container produced like the above can be a drop in replacement for a docker container where the entry point was something like `/your-binary`. If you have a more complex entrypoint, you'll need to simply preflix it with `/envoy-preflight` in the Dockerfile, most likely. And if you provide the full run command to your container, instead of just args on top of the entrypoint, you'll need to prefix that run command with `/envoy-preflight` as well.

As noted in the README, the preflight binary will do nothing (other than run your binary) if you don't provide certain env vars, at the very least `ENVOY_ADMIN_API`

**Comment by janusch** [comment]
Hi @jackkleeman 

thank you for getting back to me and for the explanation. This seems so obvious to me now looking at it ;)

In the mean time I dropped istio for a simpler routing solution that was good enough for my needs traefic.
